app/questionnaire/value_source_resolver.py

```diff
@@ -230,20 +230,20 @@ def _resolve_list_value_source(self, value_source: Mapping) -> int | str | list:
 
         return list(list_model)
 
-    def _resolve_calculated_summary_value_source(
+    def _resolve_summary_with_calculation(
         self, value_source: Mapping, *, assess_routing_path: bool
     ) -> IntOrDecimal | None:
-        """Calculates the value for the 'calculation' used by the provided Calculated Summary.
+        """Calculates the value for the 'calculation' used by the provided Calculated or Grand Calculated Summary.
 
-        The caller is responsible for ensuring the provided Calculated Summary and its answers are on the path,
+        The caller is responsible for ensuring the provided Summary and its components are on the path
```

Comment:
Summary is not really an actual block type, so shouldn't be capitalised? I can only find details like this, otherwise the code changes look good.